pkg/apis/v1alpha1/types.go

```diff
@@ -154,3 +154,54 @@ type ThanosQuerierSpec struct {
 // It should always be reconstructable from the state of the cluster and/or outside world.
 type ThanosQuerierStatus struct {
 }
+
+// MonitoringStackConfig configures the deployment of the different operators managed
+// by the Monitoring Stack Operator
+// +k8s:openapi-gen=true
+// +kubebuilder:resource
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type MonitoringStackConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MonitoringStackConfigSpec   `json:"spec,omitempty"`
+	Status MonitoringStackConfigStatus `json:"status,omitempty"`
+}
+
+// MonitoringStackConfigSpec defines the desired state of MonitoringStackConfig.
+type MonitoringStackConfigSpec struct {
+	// +kubebuilder:validation:Optional
+	MonitoringStackOperator MonitoringStackOperator `json:"monitoringStackOperator,omitempty"`
+	// +kubebuilder:validation:Optional
+	GrafanaOperator         GrafanaOperator         `json:"grafanaOperator,omitempty"`
+}
+
+// MonitoringStackConfigStatus defines the observed state of MonitoringStackConfig.
+// It should always be reconstructable from the state of the cluster and/or outside world.
+type MonitoringStackConfigStatus struct {
+}
+
+// MonitoringStackOperator exposes configuration options regarding the monitoring stack operator.
+// +kubebuilder:resource
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object 
+type MonitoringStackOperator struct {
+	// GrafanaDataSource exposes whether or not the monitoring stack should create 
+	// GrafanaDataSources or not.
+	// +kubebuilder:validation:Optional
+	GrafanaDataSource bool `json:"grafanaDataSource,omitempty"`
```

Comment:
1. For the time being I think we should create grafana datasources if the `GrafanaOperator` is enabled. 

2. Creation of grafana-datasource that is linked to a monitoring-stack should be controlled through a stack and not as global setting (IMHO)